### PR TITLE
DP-8165:  Default inputdate values.

### PR DESCRIPTION
**Pull request by smurrayatwork**
This allows default dates to be set for InputDate, which in turn will allow massgov-search to reset Pikaday dates for new searches.

Pikaday's configuration settings for formatStrict needed to be a boolean and not a string, so that was fixed as well. I also added tracking the pikaday instance local to the class.

This also adds moment.js to mayflower-react in package.json.

### CODE REVIEW

File: src/components/atoms/forms/InputDate/index.js [hunk 1]
```diff
@@ -72,13 +88,16 @@ InputDate.propTypes = {
   /** Controls whether the user can pick any date (''), today and prior ('max') or today and future ('min') */
   restrict: PropTypes.oneOf(['', 'max', 'min']),
   /** Custom onChange function that receives the selected date input */
-  onChangeCallback: PropTypes.func
+  onChangeCallback: PropTypes.func,
+  /** The date to set by default */
+  defaultDate: PropTypes.instanceOf(Date)
 };
```
Comment (avrilpearl) [inline src/components/atoms/forms/InputDate/index.js]: if reasonable, can you add this to the readme and the knobs for this components story.

File: src/components/atoms/forms/InputDate/index.js [hunk 2]
```diff
@@ -1,39 +1,55 @@
...
 import PropTypes from 'prop-types';
 import Pikaday from 'pikaday';
+import moment from 'moment';
 
 class InputDate extends React.Component {
   constructor(props) {
     super(props);
+    this.picker = null;
+    this.date = this.props.defaultDate;
     this.startPikaday = this.startPikaday.bind(this);
```
Comment (avrilpearl) [inline src/components/atoms/forms/InputDate/index.js]: should this be tracked in component state as to not cause the described issue....

File: src/components/atoms/forms/InputDate/index.js (hunk 1 above)
Comment (avrilpearl) [inline src/components/atoms/forms/InputDate/index.js]: if reasonable, can this be added to readme and knobs

**Comment by clairesunstudio** [comment]
I think for now that it's okay that the component doesn't render on prop change, but at some point we should probably update the restrict of the end-date based on the start date.

File: src/components/atoms/forms/InputDate/InputDate.stories.js [hunk 3]
```diff
@@ -17,8 +17,9 @@ storiesOf('atoms/forms', module).addDecorator(withKnobs)
       required: boolean('inputDate.required', true),
       id: text('inputDate.id', 'date-input'),
       name: text('inputDate.name', 'date-input'),
-      restrict: select('inputDate.restrict', inputDateOptions.restrict),
-      onChangeCallback: action('custom-click on select')
+      restrict: select('inputDate.restrict', inputDateOptions.restrict, ''),
+      onChangeCallback: action('custom-click on select'),
+      defaultDate: new Date(date('dateRange.endDate.defaultDate', new Date('Jan 01 2018')))
```
[the remark this answers is not in the document]
Comment (smurrayatwork) [reply src/components/atoms/forms/InputDate/InputDate.stories.js]: The knob date() function returns a string timestamp and InputDate's defaultDate uses dates (or null).
The way pikaday handles dates that are just string timestamps doesn't take into account some things that have been flagged in the issue queue (such as timezones), so wrapping it in date is better, here.

**Pull request merged**